ui/app/components/http-requests-bar-chart-small.js

```diff
@@ -0,0 +1,165 @@
+import Component from '@ember/component';
+import d3 from 'd3-selection';
+import d3Scale from 'd3-scale';
+import d3Axis from 'd3-axis';
+import d3TimeFormat from 'd3-time-format';
+import d3Transition from 'd3-transition';
+import d3Ease from 'd3-ease';
+import { assign } from '@ember/polyfills';
+import { computed } from '@ember/object';
+import { run } from '@ember/runloop';
+import { task, waitForEvent } from 'ember-concurrency';
+
+/**
+ * @module HttpRequestsContainer
+ * The HttpRequestsContainer component is the parent component of the HttpRequestsDropdown, HttpRequestsBarChart, and HttpRequestsTable components. It is used to handle filtering the bar chart and table according to selected time window from the dropdown.
+ *
+ * @example
+ * ```js
+ * <HttpRequestsContainer @counters={counters}/>
+ * ```
+ *
+ * @param counters=null {Array} - A list of objects containing the total number of HTTP Requests for each month. `counters` should be the response from the `/internal/counters/requests` endpoint which looks like:
+ * COUNTERS = [
+ *    {
+ *       "start_time": "2019-05-01T00:00:00Z",
+ *       "total": 50
+ *     }
+ * ]
+ */
+
+const HEIGHT = 125;
+const BASE_SPEED = 150;
+const DURATION = BASE_SPEED * 2;
+const UI_GRAY_300 = '#bac1cc';
+const UI_GRAY_100 = '#ebeef2';
+
+export default Component.extend({
+  classNames: ['http-requests-bar-chart-small'],
+  counters: null,
+  margin: Object.freeze({ top: 24, right: 16, bottom: 24, left: 16 }),
+  padding: 0.04,
+  width: 0,
+  height() {
+    const { margin } = this;
+    return HEIGHT - margin.top - margin.bottom;
+  },
+  parsedCounters: computed('counters', function() {
+    // parse the start times so bars and ticks display properly
+    const { counters } = this;
+    counters.reverse();
+    return counters.map((counter, index) => {
+      return assign({}, counter, {
+        start_time: d3TimeFormat.isoParse(counter.start_time),
+        fill_color: index === counters.length - 1 ? UI_GRAY_300 : UI_GRAY_100,
+      });
+    });
+  }),
+
+  yScale: computed('parsedCounters', 'height', function() {
+    const { parsedCounters } = this;
+    const height = this.height();
+    const counterTotals = parsedCounters.map(c => c.total);
+
+    return d3Scale
+      .scaleLinear()
+      .domain([0, Math.max(...counterTotals)])
+      .range([height, 0]);
+  }),
+
+  xScale: computed('parsedCounters', 'width', function() {
+    const { parsedCounters, width, margin, padding } = this;
```

Comment:
If `xScale` also depends on the margin and padding values, should they be added to the `computed` params?

Reply:
I confirmed this with Noelle, but margin and padding are variables we set ahead of time.  They don't change, and thus should not be computed, but they do affect the xScale.  Good thought, but it is good as it is.